### ISSUE TITLE
Investigate correct usage of select elements with onChange event

### ISSUE THREAD

**Issue by afercia** [Yoast/yoast-components#467]
Durine @IreneStr effort to implement `jsx-a11y` checks for the components, a specific error popped out about an `onChange` event used on a select element, in:

`composites/OnboardingWizard/components/Choice.js`

`31:5  error  onBlur must be used instead of onchange, unless absolutely necessary and it causes no negative consequences for keyboard only or screen reader users  jsx-a11y/no-onchange`

When used with a keyboard, select elements behave differently depending on the operating system. On macOS, a select always "opens" so it is possible to see all the options, use the arrow keys to select one, and confirm the choice.

Instead, on Windows it is possible to focus a select element, not open it, and use the arrow keys to select an option. In this case, if an `onChange` event is used, as soon as an arrow key is pressed, the event fires and the associated action gets executed. In most of the cases, this is not the user's intent and it produces an unexpected change of context for assistive technologies users.

For more details, see a related ticket on the WordPress trac: https://core.trac.wordpress.org/ticket/40925

To make things even more complicated, browsers have traditionally implemented teh `change` event on select elements in different ways.

What the specification says:
https://www.w3.org/TR/DOM-Level-2-Events/events.html#Events-eventgroupings-keyevents
> change
> The change event occurs --> **when a control loses the input focus and its value has been modified since gaining focus** <--. This event is valid for INPUT, SELECT, and TEXTAREA. element.

Historically browsers have implemented it in a non-conforming way, except Firefox that, for many years, fired the change event **only when a select was blurred**. However, a recent change made Firefox behave like all the other browsers. For more details, see https://bugzilla.mozilla.org/show_bug.cgi?id=1350700

The point is, when changing a select option triggers an action that causes an unexpected change of context, or an undesired action for keyboard users, then `change` shouldn't be used. [Jsx-a11y warns to use `blur` instead of `change`](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-onchange.md) to implement the expected behavior but to be 100% safe, a `<select>` element should always be associated with a button to confirm the users intent.

That said, I'm not fully sure the select from the `<Choice>` component is actually used. We should investigate if it is or not, and make a decision, maybe deprecate it.

However, in the plugin configuration wizard, there is a select element with a change event that should be improved, see https://github.com/Yoast/wordpress-seo/blob/6a08a64946be519ad6d37191103ec8a240eae45d/js/src/components/ConnectGoogleSearchConsole.js#L273
